datafusion/core/src/dataframe.rs

```diff
@@ -1007,4 +1047,37 @@ mod tests {
         .await?;
         Ok(())
     }
+
+    #[tokio::test]
+    async fn with_column() -> Result<()> {
+        let df = test_table().await?.select_columns(&["c1", "c2", "c3"])?;
+        let ctx = SessionContext::new();
+        let df_impl = Arc::new(DataFrame::new(ctx.state.clone(), &df.plan.clone()));
+
+        // check that we correctly read from the table
+        let df_results = &df_impl
+            .filter(col("c2").eq(lit(3)))?
```

Comment:
@andygrove I didn't find the way how to chain filters(AND/OR), not sure if its implemented, or has to be implemented